java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -0,0 +1,476 @@
+          } else if (!variableTree.type().symbolType().isPrimitive()) {
+            sv = SymbolicValue.NULL_LITERAL;
+          }
+          if (sv != null) {
+            programState = ProgramState.put(programState, variableTree.symbol(), sv);
+          }
+        } else {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 1);
+          programState = unstack.a;
+          programState = ProgramState.put(programState, variableTree.symbol(), unstack.b.get(0));
+        }
+        break;
+      case ASSIGNMENT:
+        AssignmentExpressionTree assignmentExpressionTree = ((AssignmentExpressionTree) tree);
+        // FIXME restricted to identifiers for now.
+        if (assignmentExpressionTree.variable().is(Tree.Kind.IDENTIFIER)) {
+          Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 2);
+          SymbolicValue value = unstack.b.get(1);
+          programState = unstack.a;
+          programState = ProgramState.put(programState, ((IdentifierTree) assignmentExpressionTree.variable()).symbol(), value);
+          programState = ProgramState.stackValue(programState, value);
+        }
+        break;
+      case ARRAY_ACCESS_EXPRESSION:
+        ArrayAccessExpressionTree arrayAccessExpressionTree = (ArrayAccessExpressionTree) tree;
+        //unstack expression and dimension
+        Pair<ProgramState, List<SymbolicValue>> unstack = ProgramState.unstack(programState, 2);
+        programState = unstack.a;
+        programState = ProgramState.stackValue(programState, constraintManager.createSymbolicValue(arrayAccessExpressionTree));
+        break;
+      case NEW_ARRAY:
+        NewArrayTree newArrayTree = (NewArrayTree) tree;
+        programState = ProgramState.unstack(programState, newArrayTree.initializers().size()).a;
+        SymbolicValue svNewArray = constraintManager.createSymbolicValue(newArrayTree);
+        programState = ProgramState.stackValue(programState, svNewArray);
+        programState = svNewArray.setSingleConstraint(programState, NullConstraint.NOT_NULL);
+        break;
+      case NEW_CLASS:
+        NewClassTree newClassTree = (NewClassTree) tree;
+        programState = ProgramState.unstack(programState, newClassTree.arguments().size()).a;
+        SymbolicValue svNewClass = constraintManager.createSymbolicValue(newClassTree);
+        programState = ProgramState.stackValue(programState, svNewClass);
+        programState = svNewClass.setSingleConstraint(programState, NullConstraint.NOT_NULL);
+        break;
+      case MULTIPLY:
+      case DIVIDE:
+      case REMAINDER:
+      case PLUS:
+      case MINUS:
+      case LEFT_SHIFT:
+      case RIGHT_SHIFT:
+      case UNSIGNED_RIGHT_SHIFT:
+      case AND:
+      case XOR:
+      case OR:
+      case GREATER_THAN:
+      case GREATER_THAN_OR_EQUAL_TO:
+      case LESS_THAN:
+      case LESS_THAN_OR_EQUAL_TO:
+      case EQUAL_TO:
+      case NOT_EQUAL_TO:
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1151)